Use /proc/meminfo instead of the size of /proc/kcore since the later
is beyond good and evil with newer kernels (2.6.32 and above as it
seems). This fixes Debian bug #564466.

Index: dphys-swapfile-20061020/dphys-swapfile
===================================================================
--- dphys-swapfile-20061020.orig/dphys-swapfile	2010-05-05 20:37:42.000000000 +0200
+++ dphys-swapfile-20061020/dphys-swapfile	2010-05-05 20:41:27.000000000 +0200
@@ -47,10 +47,8 @@
     if [ "${CONF_SWAPSIZE}" = "" ] ; then
       # compute automatic optimal size
       echo -n "computing size, "
-      # this seems to be the nearest to physical RAM size, lacks about 60k
-      KCORESIZE="`ls -l /proc/kcore | awk '{ print $5 }'`"
-      # make MBytes which rounded down will be exactly 1 too few, so add 1
-      MEMSIZE="`expr "${KCORESIZE}" / 1048576 + 1`"
+      # /proc/kcore became unreliable so let's use /proc/meminfo instead
+      MEMSIZE="`head -1 /proc/meminfo | awk '{ print $2 }'`"
       # default, without config file overwriding, swap=2*RAM
       CONF_SWAPSIZE="`expr "${MEMSIZE}" '*' "${CONF_SWAPFACTOR}"`"
     fi
